### PR TITLE
[gui] refactor favourites dialog to be a window

### PR DESCRIPTION
**Pull request by xhaggi**
This will refactor our favourite dialog to be a media window. The biggest advantage over being a dialog is that it is now part of the window history. If you start a video, music etc. favourite and back out you will end up in the favourite window not on the home screen or somewhere else. It also supports views and sorting items like all other media windows.

@Razzeee @ronie @phil65 mind taking a look.

### CODE REVIEW

File: xbmc/windows/Makefile [hunk 1]
```diff
@@ -1,5 +1,7 @@
 SRCS=GUIMediaWindow.cpp \
+	 GUIViewStateFavourites.cpp \
```
Comment (ksooo) [inline xbmc/windows/Makefile]: Haha. Major bug. Wrong intentation. ;-)

Comment (xhaggi) [inline xbmc/windows/Makefile]: already fixed as I saw it directly after PR creation ;)

**Comment by xhaggi** [comment]
@phil65 my changes to Estuary are only the required ones to get it to work properly. It would be nice if you could adjust the window if we decide to merge this PR.

**Comment by ronie** [comment]
i never use favourites myself, but i wonder if the manually sorting option (move item up/down) won't be missed.
since the favourites list contains mixed content, people may want to group items of the same content together?

**Comment by BigNoid** [comment]
As discussed on slack, -1 from me. There are many items that don't have metadata and for those there's no benefit to have them in a media window or to have those in different views. IMO favourites is not necessarily meant for playing media, we have a playlist window for that where you can order/remove/add media and play it from there. In combination with an "add to playlist" context addon it can completely replace favourites for this use case.

**Comment by NedScott** [comment]
I think this is desired even if the favs are not media items. I don't use any direct media in my favs, but I would still really like to see this included. 

For example, this makes it very nice to mix shorcuts to video nodes along side Kodi add-ons or Android apps (for video apps). I can have my own "Ned's movies", then "Netflix", then "Ned's music", which is not possible with video nodes alone. 

In Confluence I would love to have had Favs be in an icon view, as opposed to a list view. I would have also loved to have had the favs window show up in the history of the windows, so that you go back to favs after you exit out of something. This is the only universal window that users can build up, which isn't skin dependent.

With such examples, is there still an objection here?

**Comment by Rechi** [comment]
@xhaggi this needs a rebase

**Comment by xhaggi** [comment]
I'll close this PR for now, because some of our members do not like the favourites dialog to be a window.

**Pull request closed without merging**